### PR TITLE
Fix too extensive host notification recipients

### PR DESCRIPTION
**Pull request by lippserd**
Previously, host notification recipients also included notification recipients from their services.

### CODE REVIEW

**Review by nilmerg — DISMISSED** [review]

**Comment by lippserd** [comment]
Don't merge yet. Still testing whether this works.

**Comment by nilmerg** [comment]
Then make it a draft next time please :wink:

**Comment by lippserd** [comment]
> Then make it a draft next time please 😉

Sry!

**Comment by julianbrost** [comment]
Fixes the issue for me. An intermediate patch showed too little results sometimes, but I don't remember the exact conditions when this happened, so I tried a few different user/group configuration and everything showed up fine with this patch.

File: library/Icingadb/Widget/Detail/ObjectDetail.php [hunk 1]
```diff
@@ -477,16 +477,29 @@ protected function getUsersAndUsergroups(): array
...
+            );
+        }
 
         if ($this->isPermittedRoute('users')) {
             $userQuery = User::on($this->getDb());
             $userQuery->filter($objectFilter);
             $this->applyRestrictions($userQuery);
+            if ($groupBy) {
+                $userQuery->getSelectBase()->groupBy(['user.id']);
+            }
```
Comment (nilmerg) [inline library/Icingadb/Widget/Detail/ObjectDetail.php:502]: Empty lines after control structures, please.

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Pull request merged**